components/WorkSectionAsGallery.js

```diff
@@ -0,0 +1,221 @@
+        </video>
+      }
+    </div>
+  );
+}
+
+const renderWorkGalleryBlock = (workGalleryBlock, index) => {
+  const fullWidthWork = get(workGalleryBlock, 'fields.fullWidthWork', {});
+  const halfWidthWorkOne = get(workGalleryBlock, 'fields.halfWidthWorkOne', {});
+  const halfWidthWorkTwo = get(workGalleryBlock, 'fields.halfWidthWorkTwo', {});
+
+  return (
+    <div className="flex-col">
+      {workGalleryBlock.fields.displayFullWidthAssetAtTop && 
+        <div className="block col-8 relative">
+          {renderWorkOverlay(fullWidthWork)}
+          {renderWorkAsImageOrVideo(fullWidthWork)}
+        </div>
+      }
+      <div className="flex col-8 pb_5 md:pb1 items-end" key={index}>
+        <div className="block col-4 pr_25 md:pr_5">
+          <div className="relative">
+            {renderWorkOverlay(halfWidthWorkOne)}
+            {renderWorkAsImageOrVideo(halfWidthWorkOne)}
+          </div>
+        </div>
+        <div className="block col-4 pr_25 md:pr_5">
+          <div className="relative">
+            {renderWorkOverlay(halfWidthWorkTwo)}
+            {renderWorkAsImageOrVideo(halfWidthWorkTwo)}
+          </div>
+        </div>
+      </div>
+      {!workGalleryBlock.fields.displayFullWidthAssetAtTop && 
+        <div className="block col-8 relative">
+          {renderWorkOverlay(fullWidthWork)}
+          {renderWorkAsImageOrVideo(fullWidthWork)}
+        </div>
+      }
+    </div>
+  );
+}
+
+class WorkSectionAsGallery extends PureComponent {
+  render() {
+    const fields = get(this, 'props.workGallery.fields', {}); 
+    const workGalleryBlockOne = get(fields, 'workGalleryBlockOne', {});
+    const workGalleryTextOne = get(fields, 'workGalleryTextOne', '');
+    const workGalleryBlockTwo = get(fields, 'workGalleryBlockTwo', {});
+    const workGalleryTextTwo = get(fields, 'workGalleryTextTwo', '');
+    const workGalleryBlockAnyRemaining = get(fields, 'workGalleryBlockAnyRemaining', []);
+
+    return (
+      <div className="WorkSectionAsGallery">
+        {renderWorkGalleryBlock(workGalleryBlockOne)}
+        <Markdown
+          className="WorkSectionAsGallery__work-gallery-text Markdown--medium"
+          src={workGalleryTextOne}
+        />
+        {renderWorkGalleryBlock(workGalleryBlockTwo)}
+        <Markdown
+          className="WorkSectionAsGallery__work-gallery-text Markdown--medium"
+          src={workGalleryTextTwo}
+        />
+        {workGalleryBlockAnyRemaining.map((block) => renderWorkGalleryBlock(block))}
+      </div>
+    );
+  }
+}
+
+const workGalleryBlockPropType = PropTypes.shape({
+  title: PropTypes.string, 
+  fullWidthWork: PropTypes.shape({
+    fields: PropTypes.shape({
+      title: PropTypes.string,
+      caseStudySlug: PropTypes.string,
+      asset: ContentfulMedia,
+      link: PropTypes.string,
+      linkLabel: PropTypes.string,
+      stack: SimpleFragment,
+      collaborators: SimpleFragment
+    })
+  }),
```

Comment:
types for `fullWidthWork`, `halfWidthWorkOne`, and `halfWidthWorkTwo` seem repeated, i think you can just declare them once and re-use?

Reply:
faxxxx!!!!